weatherman/weatherreport.py

```diff
@@ -0,0 +1,70 @@
+"""
+For reading data and generating weather report
+"""
+import os
+from weatherobj import WeatherObj
+from utils import replace_string_in_dict
+
+
+class WeatherReport:
+    __FILE_PREFIX = "lahore_weather"
+    __KEY_DATE = "PKT"
+    __KEY_DATE_ALTER = "PKST"
+
+    def __init__(self, path_to_dir):
+        self.__path_to_dir = path_to_dir
+        self.weather_data = dict()
+
+    def __get_file_list(self):
+        return [f'.{self.__path_to_dir}/{file_name}' for file_name in os.listdir(f'.{self.__path_to_dir}')
+                if file_name.startswith(self.__FILE_PREFIX)]
+
+    def __should_ignore_reading(self, line):
+        for key in WeatherObj.KEYS_WEATHER_ATTRIB:
+            if key not in line or not line[key]:
+                return True
+        else:
+            return False
+
+    def __get_file_summary(self, name_of_file):
+        summary = None
+        with open(name_of_file) as weather_file:
+            weather_file.readline()
+            header = [item.strip() for item in weather_file.readline().split(",")]
+
+            for line in weather_file.readlines()[:len(weather_file.readlines())-1]:
```

Comment:
Use alternate way to do this rather than reading whole file just to count the number of lines. You can use `-1` index